code-styling/mysql/README.md

```diff
@@ -0,0 +1,574 @@
+```sql
+INSERT INTO my_table (column_a, column_b) VALUES ('A', 'B');
+
+INSERT INTO my_table (column_a, column_b) VALUES ('A', 'B'), ('C', 'D');
+```
+
+### Update queries
+
+Update queries MUST have a `WHERE` clause that limits the number of rows that
+will be updated. The where clause MUST NOT be indented.
+
+The `UPDATE` keyword MUST be followed by the table name that will be affected.
+The `SET` keyword MUST be on the next line and and MUST not be indented. Column
+names should be on separate lines, followed by an equals sign and the new value
+(all on the same line). If only one column will be updated, this MAY be on the
+same line as the `SET` keyword.
+
+Good update queries:
+
+```sql
+UPDATE my_table
+SET column_a = 'new_value'
+WHERE id = 5;
+
+UPDATE my_table
+SET
+  column_a = 'new_value'
+WHERE id = 5;
+
+UPDATE my_table
+SET
+  column_a = 'new_value',
+  column_b = 'other_value'
+WHERE id = 5;
+```
+
+Bad update queries:
+
+```sql
+UPDATE my_table
+SET column_a = 'new_value', column_b = 'other_value'
+WHERE id = 5;
+```
+
+### Other select criteria
+
+The `BETWEEN ... AND ...` keywords MUST be given on one line.
+
+Good usage of `BETWEEN ... AND ...`
+
+```sql
+SELECT
+  id
+FROM my_table
+WHERE id BETWEEN 10 AND 20;
+```
+
+Bad usage of `BETWEEN ... AND ...`
+
+```sql
+SELECT
+  id
+FROM my_table
+WHERE id BETWEEN 10
+AND 20;
+
+SELECT
+  id
+FROM my_table
+WHERE id BETWEEN
+10 AND 20;
+```
+
+### Grouping
+
+A `GROUP BY` clause MUST be used when aggregation functions are used. The
+`SELECT` fields SHOULD contain all columns of the `GROUP BY` clause and
+aggregated columns. Columns that are not part of the `GROUP BY` clause and not
+aggregated, SHOULD NOT appear in the field list.
+
+If aggregation functions are not used, `GROUP BY` also SHOULD NOT be used.
+Instead use `SELECT DISTINCT`.
+
+Good `GROUP BY` usage:
+
+```sql
+SELECT
+  t1.column_a,
+  MAX(t1.column_b)
+FROM table1 AS t1
+GROUP BY t1.column_a;
+
+SELECT DISTINCT
+  t1.column_a,
+  t1.column_b
+FROM table1 AS t1;
+```
+
+Bad `GROUP BY` usage:
+
+```sql
+SELECT
+  t1.column_a,
+  t1.column_b
+FROM table1 AS t1
+GROUP BY t1.column_a;
+```
+
+### Ordering
+
+Every column name in an `ORDER BY` clause MUST be followed by either `ASC` or
+`DESC` to specify the order direction.
+
+The `ORDER BY` keyword SHOULD be followed by the column name or column names on
+which to sort. If the line gets too long, column names MAY be given in a
+multiline sequence with one column per line.
+
+Good `ORDER BY` clauses:
+
+```sql
+ORDER BY table_alias.column_a DESC
+
+ORDER BY table_alias.column_a DESC, table_alias.column_b ASC
+
+ORDER BY
+  table_alias.column_a DESC,
+  table_alias.column_b ASC,
+  other_table_alias.column_c ASC
+```
+
+Bad `ORDER BY` clauses:
+
+```sql
+ORDER BY table_alias.column_a
+
+ORDER BY table_alias.column_a DESC, table_alias.column_b ASC, other_table_alias.column_c ASC
+```
+
+### Limit
+
+A `LIMIT` clause SHOULD only be used together with an `ORDER BY` clause.
+
```

Comment:
I disagree here
Offset should be used as a separate keyword instead of being a part of the limit statement, so limit 20,10 is invalid

Limit is valid without order by, just give me 10 something from the table, i do not care about the order/grouping
Even with offset it could be valid for paging, except it would be better to do it in a begin/end block.